sentinel-adapter/sentinel-zuul-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/gateway/zuul/api/route/RegexRoutePathMatcher.java

```diff
@@ -39,7 +40,13 @@ public RegexRoutePathMatcher(String pattern) {
 
     @Override
     public boolean test(RequestContext context) {
-        String path = context.getRequest().getServletPath();
+        //Solve the problem of prefix and suffix matching
+        HttpServletRequest request = context.getRequest();
+        String path = request.getServletPath();
+        String pathInfo = request.getPathInfo();
+        if (StringUtils.length(pathInfo) > 0) {
+            path = path + pathInfo;
```

Comment:
Could there be a case like pathInfo is not trimmed or pathInfo is full of spaces? Isn't apache common StringUtils `isBlank` better?

Reply:
Ok, this piece is really my incomplete consideration, thank you for correcting, and it has been fixed

Reply:
Maybe using the embedded `com.alibaba.csp.sentinel.util.StringUtil` is enough.

Reply:
Ok, this piece is really my incomplete consideration, thank you for correcting, and it has been fixed